src/components/inputs/rich-text-input/editor.js

```diff
@@ -0,0 +1,352 @@
+    }
+  }, [editor.value, renderToggleButton]);
+
+  const activeBlock =
+    (editor.value.blocks.first() && editor.value.blocks.first().type) || '';
+
+  const activeMarks = Array.from(editor.value.activeMarks).map(
+    mark => mark.type
+  );
+
+  const activeMoreStyleMarks = activeMarks.filter(activeMark =>
+    dropdownOptions.some(dropdownOption => activeMark === dropdownOption.value)
+  );
+
+  const hasBlock = type => editor.value.blocks.some(node => node.type === type);
+
+  const onClickBlock = ({ value: type }) => {
+    const { value } = editor;
+    const { document } = value;
+
+    // Handle everything but list buttons.
+    if (type !== BULLETED_LIST && type !== NUMBERED_LIST) {
+      const isActive = hasBlock(type);
+      const isList = hasBlock(LIST_ITEM);
+
+      if (isList) {
+        editor
+          .setBlocks(isActive ? DEFAULT_NODE : type)
+          .unwrapBlock(BULLETED_LIST)
+          .unwrapBlock(NUMBERED_LIST);
+      } else {
+        editor.setBlocks(isActive ? DEFAULT_NODE : type);
+      }
+    } else {
+      // Handle the extra wrapping required for list buttons.
+      const isList = hasBlock(LIST_ITEM);
+      const isType = value.blocks.some(block => {
+        return !!document.getClosest(block.key, parent => parent.type === type);
+      });
+
+      if (isList && isType) {
+        editor
+          .setBlocks(DEFAULT_NODE)
+          .unwrapBlock(BULLETED_LIST)
+          .unwrapBlock(NUMBERED_LIST);
+      } else if (isList) {
+        editor
+          .unwrapBlock(type === BULLETED_LIST ? NUMBERED_LIST : BULLETED_LIST)
+          .wrapBlock(type);
+      } else {
+        editor.setBlocks(LIST_ITEM).wrapBlock(type);
+      }
+    }
+  };
+
+  const { toggleMark } = editor;
+
+  const onChangeMoreStyles = React.useCallback(
+    val => {
+      if (!editor.value.selection.isFocused) {
+        editor.focus();
+      }
+      toggleMark(val.value);
+    },
+    [toggleMark, editor]
+  );
+
+  const hasUndos = editor.hasUndos();
+  const hasRedos = editor.hasRedos();
+
+  return (
+    <CollapsibleMotion
+      minHeight={32}
+      isDefaultClosed={!props.defaultExpandMultilineText}
+    >
+      {({ isOpen, toggle, containerStyles, registerContentNode }) => {
+        return (
+          <Constraints.Horizontal constraint={props.horizontalConstraint}>
+            <Spacings.Stack scale="xs">
+              <Container
+                {...props}
+                onFocus={() => {
+                  if (!isOpen) {
+                    toggle();
+                    setTimeout(() => setRenderToggleButton(true), 0);
+                  }
+                }}
+              >
+                <Toolbar {...props} isOpen={isOpen}>
+                  <ToolbarMainControls>
+                    <StyleDropdown
+                      value={activeBlock}
+                      onChange={onClickBlock}
+                    />
+                    <Tooltip
+                      title="Bold"
```

Comment:
I guess those labels need to be translated?

Reply:
Yep, nice catch 👍